libsolidity/ast/Types.cpp

```diff
@@ -362,6 +368,69 @@ MemberList::MemberMap Type::boundFunctions(Type const& _type, ASTNode const& _sc
 	if (auto refType = dynamic_cast<ReferenceType const*>(&_type))
 		typeLocation = refType->location();
 
+	return usingForDirectives | ranges::views::filter([&](UsingForDirective const* _directive) -> bool {
+		// Convert both types to pointers for comparison to see if the `using for`
+		// directive applies.
+		// Further down, we check more detailed for each function if `_type` is
+		// convertible to the function parameter type.
+		return
+			!_directive->typeName() ||
+			*TypeProvider::withLocationIfReference(typeLocation, &_type, true) ==
+			*TypeProvider::withLocationIfReference(
+				typeLocation,
+				_directive->typeName()->annotation().type,
+				true
+			);
+	}) | ranges::to<vector<UsingForDirective const*>>;
+}
+
+}
+
+Result<FunctionDefinition const*> Type::userDefinedOperator(Token _token, ASTNode const& _scope, bool _unaryOperation) const
+{
+	if (!typeDefinition())
+		return nullptr;
+
+	set<FunctionDefinition const*> seenFunctions;
+	for (UsingForDirective const* ufd: usingForDirectivesForType(*this, _scope))
+		for (auto const& [pathPointer, operator_]: ufd->functionsAndOperators())
```

Comment:
I find it more readable when it's simply called `identifierPath`. I had to recheck what the type actually was. And we usually don't differentiate between pointers, references and values in names:
```suggestion
		for (auto const& [identifierPath, operator_]: ufd->functionsAndOperators())
```

Reply:
done

Reply:
There's one more instance, in `Type::boundFunctions()`.